### PR TITLE
Add tests for Cluster Resource attributes

### PR DESCRIPTION
**Pull request by anpanigr**
Add more tests for RestartVersion, ServerStartPolicy attributes
Add tests for Configured Cluster

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13462

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiClusterResource.java [hunk 1]
```diff
@@ -714,6 +722,115 @@ void testKubectlScaleClusterResource() {
...
+   * Create a WebLogic domain resource DR with domain level replica set to zero.
+   * Create and deploy two cluster resources CR1 and CR2
+   * Create and deploy the domain with two cluster resources CR1 and CR2
+   * Restart only the cluster CR2 by incrementing restart version on 
+   * Cluster resource and make sure no effect on CR1 
+   * Stop the cluster CR1 by updating the serverStartPolicy to Never
+   * Start the cluster CR1 by updating the serverStartPolicy to IfNeeded
+   */
+  @Test
+  @DisplayName("Verify various kubectl scale options")
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiClusterResource.java:735]: fix description

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Cluster.java [hunk 2]
```diff
@@ -102,6 +104,68 @@ public static boolean scaleCluster(String clusterRes, String namespace, int numO
...
+   */
+  public static String patchClusterResourceWithNewRestartVersion(
+      String clusterResourceName, String namespace) {
+
+    LoggingFacade logger = getLogger();
+    String oldVersion = assertDoesNotThrow(
+        () -> getClusterCustomResource(clusterResourceName, namespace, CLUSTER_VERSION).getSpec().getRestartVersion(),
+        String.format("Failed to get the restartVersion of %s in namespace %s", clusterResourceName, namespace));
+    int newVersion = oldVersion == null ? 1 : Integer.valueOf(oldVersion) + 1;
+    logger.info("Update domain resource {0} in namespace {1} restartVersion from {2} to {3}",
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Cluster.java:121]:
```suggestion
    logger.info("Update cluster resource {0} in namespace {1} restartVersion from {2} to {3}",
```

**Review by vanajamukkara — COMMENTED** [review]
left comments for minor changes

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Cluster.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Cluster.java]: done

**Review by vanajamukkara — APPROVED** [review]

**Review by doxiao — APPROVED** [review]

**Pull request merged**